util/devel/deprecationScript/extractLineNumbers.py

```diff
@@ -0,0 +1,101 @@
+import re
+import sys
+import os
+
+TEST_LINE_PREFIX = '[test: '
+TEST_LINE_CLOSING = ']'
+CHPL_REGEX_PATTERN = r'.{0,3}\$CHPL_HOME/'
+DIR_ABS_PREFIX = '$CHPL_HOME/test/'
+CHPL_EXT = '.chpl'
+DIFF_INDICATOR = '> '
+
+def extractLineNumbers(file_name, warning_message, output_file_name):
+  # Read in the text file
+  with open(file_name, 'r') as f:
+    lines = f.readlines()
+  
+
+  # Extract lines with a specific characteristic
+  warning_lines = []
+  current_sub_test_path = None
+
+  for line in lines:
+    # print(line)
+    # If the line begins with "[test: " then we're running a new subtest"
+    if line.startswith(TEST_LINE_PREFIX):
+      # Extract the current_sub_test_path from the line 
+      # excluding the filename at the end of the path
+      # Ex: [test: attributes/badAttribute.chpl] -> path = attributes/
+      current_sub_test_path = line.split(TEST_LINE_CLOSING)[0].split(TEST_LINE_PREFIX)[1].rsplit('/', 1)[0] + '/'
+    # print(current_sub_test_path)
+
+    if warning_message in line:
+      warning_line = resolveToAbsPath(current_sub_test_path, line)
+      if warning_line is not None:
+        warning_lines.append(warning_line)
+
+  # Remove duplicates efficiently
+  warning_lines = list(dict.fromkeys(warning_lines))
+
+  # Write them to a file
+  with open(output_file_name, 'w') as f:
+    for line in warning_lines:
+      f.write(line)
+  print("Total Unique Matches Found: " + str(len(warning_lines)))
+  return
+
+def resolveToAbsPath(current_sub_test_path, line):
+
+  # If the error comes from an internal chapel module it will be an absolute path already
+  # The path will be something like '$CHPL_HOME/modules/internal/' 
+  # We do a lose check to see if the beginning of the line approximately matches such a path
+  # we use regex for an approximate match
+  # Create the regex with room for 3 characters before matching starts
+  # This is because the line will start with '> $CHPL_HOME'
+  # We don't want to match the first 0 to 3 characters of the path
+  # Ex: '> $CHPL_HOME/modules/internal/' -> '$CHPL_HOME/modules/internal/'
+  regex = re.compile(CHPL_REGEX_PATTERN)
+  if regex.match(line):
```

Comment:
There is no real reason to compile a regex and then only use it once, just do `re.match(CHPL_REGEX_PATTERN, line)`. You only get a benefit if you precompile the regex and then use it multiple times. So in that case, I would hoist the compiled regex to be near the definition of `CHPL_REGEX_PATTERN` (and maybe put the comment about how the regex is put together there)

Reply:
Got it thanks! I'll use `re.match(CHPL_REGEX_PATTERN, line)`